src/components/renderer/PageFormElements.tsx

```diff
@@ -39,45 +39,63 @@ function PageFormElements({
   const handleLookup = React.useCallback<FormElementLookupHandler>(
     (mergeLookupResults) => {
       setFormSubmission((currentFormSubmission) => {
-        const { submission, elements, executedLookups } = mergeLookupResults({
-          elements: pageElement.elements,
-          submission: currentFormSubmission.submission,
-          lastElementUpdated: currentFormSubmission.lastElementUpdated,
-          executedLookups: currentFormSubmission.executedLookups,
-        })
-
-        const definition = {
-          ...currentFormSubmission.definition,
-        }
         if (pageElement.id === formId.toString()) {
-          definition.elements = elements
-        } else {
-          definition.elements = currentFormSubmission.definition.elements.map(
-            (formElement) => {
-              if (
-                formElement.id === pageElement.id &&
-                formElement.type === 'page'
-              ) {
-                return {
-                  ...formElement,
-                  elements,
-                }
-              } else {
-                return formElement
-              }
+          const { submission, elements, executedLookups } = mergeLookupResults({
+            elements: currentFormSubmission.definition.elements,
```

Comment:
Won't this cause some problems? Because previously we were passing page elements, now its just the root elements array.